### PR TITLE
Add org.neverball.Neverball

### PR DESCRIPTION
**Pull request by AsciiWolf**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/Neverball/neverball/issues/193#issuecomment-761556284
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
It is the [official](https://neverball.org) Neverball domain.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. If not, explain why:
The last Neverball version was released in 2014 and does not contain any AppStream metadata file. However, an AppData file [is present](https://github.com/Neverball/neverball/blob/master/dist/neverball.appdata.xml) in the Neverball Git repo. The only changes that I made are adding an information about Neverputt (that are useful for this Flatpak that has Neverputt bundled (and splitting Neverputt into a separate Flatpak would not make much sense since the same game data files and game config directory are used by both Neverball and Neverputt), but not useful for upstream AppData file that is used by some downstream distribution packages that do not contain Neverputt) and release information that also are not useful in upstream (without being regularly updated). The two patch files used in this Flatpak are just workarounds for issues already fixed in devel Git version of Neverball.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by AsciiWolf** [comment]
bot, build org.neverball.Neverball

File: org.neverball.Neverball.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    sources:
+      - type: archive
+        url: https://github.com/Neverball/neverball/archive/neverball-1.6.0.tar.gz
+        sha256: 737377830aaa2069e00e2e5e4434c5578efda40d5dacb38d715643e52d9753bc
+      - type: patch
+        paths:
+          - makefile.patch
+          - version.patch
+      - type: file
+        path: org.neverball.Neverball.appdata.xml
```
Comment (LeandroStanger) [inline org.neverball.Neverball.yaml:33]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by LeandroStanger** [comment]
bot, build org.neverball.Neverball.appdata

File: org.neverball.Neverball.yaml (hunk 1 above)
Comment (AsciiWolf) [reply org.neverball.Neverball.yaml:33]: I think this does not matter that much. Even GNOME Software itself still use appdata and when I use `/app/share/metainfo` directory, flatpak-builder moves the appdata file to `/app/share/appdata`.

**Comment by AsciiWolf** [comment]
> bot, build org.neverball.Neverball.appdata

You did not have to run the build again, the previous one finished few minutes ago and there were no changes since then. ;-)

**Comment by AsciiWolf** [comment]
bot, build org.neverball.Neverball

File: org.neverball.Neverball.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+sdk: org.freedesktop.Sdk
+command: neverball
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --persist=.neverball
+modules:
+  - name: physfs
```
Comment (barthalion) [inline org.neverball.Neverball.yaml:13]: Misses `cleanup` to delete static library and headers.

Comment (AsciiWolf) [reply org.neverball.Neverball.yaml:13]: Oops, I forgot adding it. Thanks for reminding me! :-)

**Comment by AsciiWolf** [comment]
bot, build org.neverball.Neverball

**Pull request closed without merging**